lib/zclient.c

```diff
@@ -1326,6 +1326,26 @@ enum zclient_send_status zclient_nhg_send(struct zclient *zclient, int cmd,
 	return zclient_send_message(zclient);
 }
 
+/* size needed by a stream for redistributing a route */
+int zapi_redistribute_stream_size(struct zapi_route *api)
+{
+
+	size_t msg_size = 0;
+	size_t nh_size = sizeof(struct zapi_nexthop);
+
+	msg_size =  sizeof(struct zapi_route);
+	/* remove unused nexthop structures */
+	msg_size += (MULTIPATH_NUM - api->nexthop_num) * nh_size;
```

Comment:
comment says "remove", but ... these are adding?
don't you want to reduce 'msg_size' by the number of nexthops that were counted in the zapi_route struct, but that are not being sent?